optimade/validator/validator.py

```diff
@@ -483,32 +489,36 @@ def _construct_queries_for_property(
                 f"Chosen entry of endpoint '/{endp}' had unexpected type '{chosen_entry['type']}'."
             )
 
-        if prop == "id":
-            prop_type = DataType.STRING
+        if prop_type is None:
+            prop_type = CONF.entry_schemas[endp].get(prop, {}).get("type")
 
         if prop_type is None:
             raise ResponseError(
-                f"Cannot validate queries on {prop} as field type was not reported in /info/{endp}"
+                f"Cannot validate queries on {prop!r} as field type was not reported in `/info/{endp}`"
             )
 
         # this is the case of a provider field
-        if prop not in (
-            CONF.entry_properties[endp]["MUST"]
-            | CONF.entry_properties[endp]["SHOULD"]
-            | CONF.entry_properties[endp]["OPTIONAL"]
-        ):
+        if prop not in CONF.entry_schemas[endp]:
             if self.provider_prefix is None:
                 raise ResponseError(
-                    f"Found unknown field {prop} and no provider prefix was provided in `/info`"
+                    f"Found unknown field {prop!r} in `/info/{endp}` and no provider prefix was provided in `/info`"
                 )
             elif not prop.startswith(f"_{self.provider_prefix}"):
```

Comment:
Non-necessary update, but since the full provider-specific property prefix is the prefix wrapped in `_`, I thought I'd add it to be very explicit :)
```suggestion
            elif not prop.startswith(f"_{self.provider_prefix}_"):
```